src/api/spec/controllers/webui/users/tokens/groups_controller_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+RSpec.describe Webui::Users::Tokens::GroupsController do
+  let(:token_group) { create(:group_with_user) }
+  let(:other_group) { create(:group) }
+  let(:workflow_token) { create(:workflow_token) }
+  let(:token_user) { token_group.users.first }
+
+  before do
+    Flipper.enable(:trigger_workflow, token_user)
+    login token_user
+    workflow_token.groups_shared_among << token_group
+  end
+
+  describe '#create' do
+    render_views
+
+    context 'when the group does not own the token yet' do
+      subject! do
+        post :create, params: { groupid: other_group, token_id: workflow_token }
+      end
+
+      it { is_expected.to redirect_to(token_users_path(workflow_token)) }
+      it { expect(flash[:success]).to be_present }
+      it { expect(workflow_token.groups_shared_among).to match_array([token_group, other_group]) }
+    end
+
+    context 'when the group owns the token already' do
+      before do
+        workflow_token.groups_shared_among << other_group
+      end
+
+      subject! do
+        post :create, params: { groupid: other_group, token_id: workflow_token }
+      end
+
+      it { is_expected.to redirect_to(token_users_path(workflow_token)) }
+      it { expect(flash[:success]).to be_present }
+      it { expect(workflow_token.groups_shared_among).to match_array([token_group, other_group]) }
+    end
```

Comment:
You have no logic for this in the controller anymore. Remove the test, CI time is developer time :)